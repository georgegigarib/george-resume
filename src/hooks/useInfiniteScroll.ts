import { useEffect, useRef } from 'react'

const useInfiniteScroll = (containerId: string) => {
  const containerRef = useRef<HTMLDivElement>(null)

  useEffect(() => {
    const container = containerRef.current
    if (!container) return

    let animationFrame: number
    let scrollPosition = 0

    try {
      // Duplicamos el contenido para permitir un scroll infinito
      const originalContent = container.innerHTML
      container.innerHTML += originalContent

      // Calculamos el ancho total para ajustar el desplazamiento
      const scrollWidth = container.scrollWidth / 2

      const scrollStep = () => {
        scrollPosition += 0.5 // Ajusta la velocidad del desplazamiento aquÃ­
        if (scrollPosition >= scrollWidth) {
          scrollPosition = 0
        }
        container.scrollLeft = scrollPosition
        animationFrame = requestAnimationFrame(scrollStep)
      }

      scrollStep()
    } catch (error) {
      console.error('Error in useInfiniteScroll:', error)
    }

    // Limpieza al desmontar
    return () => {
      if (animationFrame) cancelAnimationFrame(animationFrame)
    }
  }, [containerId])

  return containerRef
}

export default useInfiniteScroll
