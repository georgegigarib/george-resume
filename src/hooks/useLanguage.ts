import { useTranslation } from 'react-i18next';

export const useLanguage = () => {
  const { i18n } = useTranslation();

  const changeLanguage = (lng: string) => {
    i18n.changeLanguage(lng);
    localStorage.setItem('language', lng);
  };

  const loadLanguageFromStorage = () => {
    const savedLanguage = localStorage.getItem('language') || 'en';
    i18n.changeLanguage(savedLanguage);
  };

  return { changeLanguage, loadLanguageFromStorage };
};
