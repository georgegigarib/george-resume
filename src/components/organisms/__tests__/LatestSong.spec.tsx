import { render, screen, act } from '@testing-library/react';
import { describe, it, expect, vi, Mock } from 'vitest';
import { useIsMobile } from '@/hooks/useIsMobile';
import LatestSong from '../LatestSong';

vi.mock('@/hooks/useIsMobile', () => ({
  useIsMobile: vi.fn(),
}));

describe('LatestSong Component', () => {
  const renderComponent = () =>
    render(<LatestSong />);

  it('matches snapshot at initial render', () => {
    (useIsMobile as Mock).mockReturnValue(false);

    const { asFragment } = renderComponent();

    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot after 5 seconds', async () => {
    vi.useFakeTimers();
    (useIsMobile as Mock).mockReturnValue(false);

    const { asFragment } = renderComponent();

    act(() => {
      vi.advanceTimersByTime(5000);
    });

    expect(asFragment()).toMatchSnapshot();

    expect(screen.getByTestId('song-test-id-1')).toBeVisible();
  });

  it('matches snapshot after another 5 seconds', async () => {
    vi.useFakeTimers();
    (useIsMobile as Mock).mockReturnValue(false);

    const { asFragment } = renderComponent();

    act(() => {
      vi.advanceTimersByTime(10000);
    });

    expect(asFragment()).toMatchSnapshot();

    expect(screen.getByTestId('song-test-id-0')).toBeVisible();
  });
});
