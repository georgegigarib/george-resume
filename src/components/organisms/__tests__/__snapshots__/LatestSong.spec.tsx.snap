// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LatestSong Component > matches snapshot after 5 seconds 1`] = `
<DocumentFragment>
  <div
    class="overflow-hidden relative"
    data-testid="frame-container"
  >
    <div
      class="transition-transform duration-500"
      style="transform: translateY(-100%); height: 152px;"
    >
      <div
        style="pointer-events: none;"
      >
        <iframe
          allow="autoplay; clipboard-write; encrypted-media *; fullscreen; picture-in-picture"
          class="rounded-xl"
          data-testid="song-test-id-0"
          height="152"
          loading="lazy"
          sandbox="allow-forms allow-popups allow-same-origin allow-scripts allow-storage-access-by-user-activation allow-top-navigation-by-user-activation"
          src="https://open.spotify.com/embed/album/6H73t0Wok8dsEEGqRhgOsV?utm_source=generator"
          style="pointer-events: auto; overflow: hidden; background: transparent;"
          width="365"
        />
      </div>
      <div
        style="pointer-events: none;"
      >
        <iframe
          allow="autoplay; clipboard-write; encrypted-media *; fullscreen; picture-in-picture"
          class="rounded-xl"
          data-testid="song-test-id-1"
          height="152"
          loading="lazy"
          sandbox="allow-forms allow-popups allow-same-origin allow-scripts allow-storage-access-by-user-activation allow-top-navigation-by-user-activation"
          src="https://embed.music.apple.com/do/album/un-a%C3%B1o-single/1760344218"
          style="pointer-events: auto; overflow: hidden; background: transparent;"
          width="365"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`LatestSong Component > matches snapshot after another 5 seconds 1`] = `
<DocumentFragment>
  <div
    class="overflow-hidden relative"
    data-testid="frame-container"
  >
    <div
      class="transition-transform duration-500"
      style="transform: translateY(-0%); height: 152px;"
    >
      <div
        style="pointer-events: none;"
      >
        <iframe
          allow="autoplay; clipboard-write; encrypted-media *; fullscreen; picture-in-picture"
          class="rounded-xl"
          data-testid="song-test-id-0"
          height="152"
          loading="lazy"
          sandbox="allow-forms allow-popups allow-same-origin allow-scripts allow-storage-access-by-user-activation allow-top-navigation-by-user-activation"
          src="https://open.spotify.com/embed/album/6H73t0Wok8dsEEGqRhgOsV?utm_source=generator"
          style="pointer-events: auto; overflow: hidden; background: transparent;"
          width="365"
        />
      </div>
      <div
        style="pointer-events: none;"
      >
        <iframe
          allow="autoplay; clipboard-write; encrypted-media *; fullscreen; picture-in-picture"
          class="rounded-xl"
          data-testid="song-test-id-1"
          height="152"
          loading="lazy"
          sandbox="allow-forms allow-popups allow-same-origin allow-scripts allow-storage-access-by-user-activation allow-top-navigation-by-user-activation"
          src="https://embed.music.apple.com/do/album/un-a%C3%B1o-single/1760344218"
          style="pointer-events: auto; overflow: hidden; background: transparent;"
          width="365"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`LatestSong Component > matches snapshot at initial render 1`] = `
<DocumentFragment>
  <div
    class="overflow-hidden relative"
    data-testid="frame-container"
  >
    <div
      class="transition-transform duration-500"
      style="transform: translateY(-0%); height: 152px;"
    >
      <div
        style="pointer-events: none;"
      >
        <iframe
          allow="autoplay; clipboard-write; encrypted-media *; fullscreen; picture-in-picture"
          class="rounded-xl"
          data-testid="song-test-id-0"
          height="152"
          loading="lazy"
          sandbox="allow-forms allow-popups allow-same-origin allow-scripts allow-storage-access-by-user-activation allow-top-navigation-by-user-activation"
          src="https://open.spotify.com/embed/album/6H73t0Wok8dsEEGqRhgOsV?utm_source=generator"
          style="pointer-events: auto; overflow: hidden; background: transparent;"
          width="365"
        />
      </div>
      <div
        style="pointer-events: none;"
      >
        <iframe
          allow="autoplay; clipboard-write; encrypted-media *; fullscreen; picture-in-picture"
          class="rounded-xl"
          data-testid="song-test-id-1"
          height="152"
          loading="lazy"
          sandbox="allow-forms allow-popups allow-same-origin allow-scripts allow-storage-access-by-user-activation allow-top-navigation-by-user-activation"
          src="https://embed.music.apple.com/do/album/un-a%C3%B1o-single/1760344218"
          style="pointer-events: auto; overflow: hidden; background: transparent;"
          width="365"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
